File: frontend/src/services/api/analysis.ts
import { analysisRead } from "@/models/analysis";
import { UUID } from "@/models/base";
import { BaseApi } from "./base";

const api = new BaseApi();
const endpoint = "/analysis/";

export const Analysis = {
  read: (uuid: UUID): Promise<analysisRead> => api.read(`${endpoint}${uuid}`),
};

File: gui_api/app/api/routes/analysis.py
from fastapi import APIRouter
from uuid import UUID

from api import db_api
from api.routes import helpers
from api_models.analysis import AnalysisRead


router = APIRouter(
    prefix="/analysis",
    tags=["Analysis"],
)


#
# READ
#


def get_analysis(uuid: UUID):
    return db_api.get(path=f"/analysis/{uuid}")


helpers.api_route_read(router, get_analysis, AnalysisRead)
